src/storage-client/src/controller.rs

```diff
@@ -2281,19 +2492,37 @@ pub struct ExportState<T> {
     /// Description with which the export was created
     pub description: ExportDescription<T>,
 
+    /// The capability (hold on the since) that this export needs from its
+    /// dependencies (inputs). When the upper of the export changes, we
+    /// downgrade this, which in turn downgrades holds we have on our
+    /// dependencies' sinces.
+    pub read_capability: Antichain<T>,
+
+    /// The policy to use to downgrade `self.read_capability`.
+    pub read_policy: ReadPolicy<T>,
```

Comment:
Assuming that we trust the sink to hold back its own reported write frontier, we can get rid of this, yeah? It seems like the one bit that doesn't have an equivalent on `CollectionState`... the `read_policy` there means something else entirely IIUC.

Reply:
If we subscribe to the idea that sinks are just external materialized views (I do) then handling them as just another collection unifies this handling.

Regardless of this however the messages that are coming back from `clusterd` regarding sink progress should be the pristine write frontier of the sink and if there is a sink specific minus one that we want to do for some reason then this is a controller level concern (just like every other type of object). Also, reporting `F-1` means that `Antichain::from_elem(T::minimum())` has dual meaning, both that nothing is persisted and that timestamp 0 is persisted which is another reason I think we shouldn't manipulate the write frontier